recipes-kernel/linux/files/patches-5.10/0220-arm64-dts-ti-Definitions-for-runtime-pinmuxing.patch

```diff
@@ -0,0 +1,754 @@
++		>;
++	};
++
++	d6_gpio_pulldown: d6-gpio-pulldown {
++		pinctrl-single,pins = <
++			/* (AH16) GPIO0_38 */
++			AM65X_IOPAD(0x0098, PIN_INPUT_PULLDOWN, 7)
++		>;
++	};
++
++	d7_ehrpwm3_a: d7-ehrpwm3-a {
++		pinctrl-single,pins = <
++			/* (AH15) EHRPWM3_A */
++			AM65X_IOPAD(0x00AC, PIN_OUTPUT, 5)
++		>;
++	};
++
++	d7_gpio: d7-gpio {
++		pinctrl-single,pins = <
++			/* (AH15) GPIO0_43 */
++			AM65X_IOPAD(0x00AC, PIN_INPUT, 7)
++		>;
++	};
++
++	d7_gpio_pullup: d7-gpio-pullup {
++		pinctrl-single,pins = <
++			/* (AH15) GPIO0_43 */
++			AM65X_IOPAD(0x00AC, PIN_INPUT_PULLUP, 7)
++		>;
++	};
++
++	d7_gpio_pulldown: d7-gpio-pulldown {
++		pinctrl-single,pins = <
++			/* (AH15) GPIO0_43 */
++			AM65X_IOPAD(0x00AC, PIN_INPUT_PULLDOWN, 7)
++		>;
++	};
++
++	d8_ehrpwm4_a: d8-ehrpwm4-a {
++		pinctrl-single,pins = <
++			/* (AG15) EHRPWM4_A */
++			AM65X_IOPAD(0x00C0, PIN_OUTPUT, 5)
++		>;
++	};
++
++	d8_gpio: d8-gpio {
++		pinctrl-single,pins = <
++			/* (AG15) GPIO0_48 */
++			AM65X_IOPAD(0x00C0, PIN_INPUT, 7)
++		>;
++	};
++
++	d8_gpio_pullup: d8-gpio-pullup {
++		pinctrl-single,pins = <
++			/* (AG15) GPIO0_48 */
++			AM65X_IOPAD(0x00C0, PIN_INPUT_PULLUP, 7)
++		>;
++	};
++
++	d8_gpio_pulldown: d8-gpio-pulldown {
++		pinctrl-single,pins = <
++			/* (AG15) GPIO0_48 */
++			AM65X_IOPAD(0x00C0, PIN_INPUT_PULLDOWN, 7)
++		>;
++	};
++
++	d9_ehrpwm5_a: d9-ehrpwm5-a {
++		pinctrl-single,pins = <
++			/* (AD15) EHRPWM5_A */
++			AM65X_IOPAD(0x00CC, PIN_OUTPUT, 5)
++		>;
++	};
++
++	d9_gpio: d9-gpio {
++		pinctrl-single,pins = <
++			/* (AD15) GPIO0_51 */
++			AM65X_IOPAD(0x00CC, PIN_INPUT, 7)
++		>;
++	};
++
++	d9_gpio_pullup: d9-gpio-pullup {
++		pinctrl-single,pins = <
++			/* (AD15) GPIO0_51 */
++			AM65X_IOPAD(0x00CC, PIN_INPUT_PULLUP, 7)
++		>;
++	};
++
++	d9_gpio_pulldown: d9-gpio-pulldown {
++		pinctrl-single,pins = <
++			/* (AD15) GPIO0_51 */
++			AM65X_IOPAD(0x00CC, PIN_INPUT_PULLDOWN, 7)
++		>;
++	};
++
+ 	main_uart1_pins_default: main-uart1-pins-default {
+ 		pinctrl-single,pins = <
+ 			AM65X_IOPAD(0x0174, PIN_INPUT,  6)  /* (AE23) UART1_RXD */
+@@ -346,17 +956,6 @@ AM65X_IOPAD(0x02c0, PIN_OUTPUT, 0)  /* (AC8) USB1_DRVVBUS */
+ 		>;
+ 	};
+ 
+-	arduino_io_d4_to_d9_pins_default: arduino-io-d4-to-d9-pins-default {
+-		pinctrl-single,pins = <
+-			AM65X_IOPAD(0x0084, PIN_OUTPUT, 7)  /* (AG18) GPIO0_33 */
+-			AM65X_IOPAD(0x008C, PIN_OUTPUT, 7)  /* (AF17) GPIO0_35 */
+-			AM65X_IOPAD(0x0098, PIN_OUTPUT, 7)  /* (AH16) GPIO0_38 */
+-			AM65X_IOPAD(0x00AC, PIN_OUTPUT, 7)  /* (AH15) GPIO0_43 */
+-			AM65X_IOPAD(0x00C0, PIN_OUTPUT, 7)  /* (AG15) GPIO0_48 */
+-			AM65X_IOPAD(0x00CC, PIN_OUTPUT, 7)  /* (AD15) GPIO0_51 */
+-		>;
+-	};
+-
+ 	dss_vout1_pins_default: dss-vout1-pins-default {
+ 		pinctrl-single,pins = <
+ 			AM65X_IOPAD(0x0000, PIN_OUTPUT, 1)  /* VOUT1_DATA0 */
+@@ -477,14 +1076,7 @@ &main_uart2 {
+ 	status = "disabled";
+ };
+ 
+-&mcu_uart0 {
+-	pinctrl-names = "default";
+-	pinctrl-0 = <&arduino_uart_pins_default>;
+-};
```

Comment:
Is this change related?

Reply:
Yes, this was intended.
Maybe a short explanation:

I had to add all possible pinmux states for each pin into the related pinctrl node. Otherwise these states are not visible for the pinctrl node. 

This affected pins that have already been configured and assigned to interfaces like I2C, SPI and UART. Those pins and their states had to be moved to the pinctrl node. 
Otherwise we will have ownership issues. 
The kernel would complain about pins that have already been assigned to the UART device in this specific case.